### PR TITLE
[SYNCOPE-808] Added Netbeans Plugin to SYNCOPE 2_0_X

### PR DESCRIPTION
**Pull request by tmess567**
Please have a look

### CODE REVIEW

File: ide/netbeans/pom.xml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    <properties>
+      <rootpom.basedir>${basedir}/../..</rootpom.basedir>
+    </properties>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>nbm-maven-plugin</artifactId>
+                <version>${nbm-maven-plugin-version}</version>
```
Comment (ilgrosso) [inline ide/netbeans/pom.xml]: This should be declared in the `<plugin-management>` section of the root POM and then put here without version.

File: ide/netbeans/pom.xml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                <extensions>true</extensions>
+                <configuration>
+                    <useOSGiDependencies>false</useOSGiDependencies>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <version>${maven-jar-plugin-version}</version>
```
Comment (ilgrosso) [inline ide/netbeans/pom.xml]: This should be declared in the `<plugin-management>` section of the root POM and then put here without version.

File: ide/netbeans/pom.xml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.netbeans.api</groupId>
+            <artifactId>org-netbeans-api-annotations-common</artifactId>
+            <version>${org-netbeans-api-version}</version>
```
Comment (ilgrosso) [inline ide/netbeans/pom.xml]: This should be declared in the `<dependencyManagement>` section of the root POM and then put here without version.

File: ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/connector/ResourceConnector.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+public final class ResourceConnector {
+
+    private ResourceConnector() {
+    }
+
+    private static MailTemplateManagerService MAIL_TTEMPLATE_MANAGER_SERVICE;
+    private static ReportTemplateManagerService REPORT_TEMPLATE_MANAGER_SERVICE;
+    
+    public static MailTemplateManagerService getMailTemplateManagerService() throws IOException {
+        if (MAIL_TTEMPLATE_MANAGER_SERVICE == null) {
```
Comment (ilgrosso) [inline ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/connector/ResourceConnector.java]: Is this thead safe?

File: ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/view/ResourceExplorerTopComponent.java [hunk 5]
```diff
@@ -0,0 +1,550 @@
...
+
+    private void openReportEditor(final String name) throws IOException {
+        String type;
+        String content;
+        InputStream is;
+        Object format = JOptionPane.showInputDialog(null, "Select File Format",
+                "File format", JOptionPane.QUESTION_MESSAGE, null,
+                PluginConstants.REPORT_TEMPLATE_FORMATS, "TEXT");
+
+        if (format.equals("HTML")) {
```
Comment (ilgrosso) [inline ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/view/ResourceExplorerTopComponent.java]: Any chance to use the proper enum here?

**Comment by ilgrosso** [comment]
hi @tmess567 chances you can take a look at the comments and fix anytime soon?

**Comment by tmess567** [comment]
I'm kinda busy with a family function.. Will work on the suggested fixes in a day or two

**Comment by ilgrosso** [comment]
> @tmess567 commented 14 days ago
> Will work on the suggested fixes in a day or two

Guess you are hardly finding some time to work on this task, right?

**Comment by tmess567** [comment]
Busy with end semester examinations and a major project, but I'll try to find more time for this as well. @ilgrosso you weren't at the irc today, I have some questions about the thread safe comment. Please contact me when you're on.

File: ide/netbeans/pom.xml [hunk 6]
```diff
@@ -63,99 +61,79 @@ under the License.
     <dependencies>
         <dependency>
             <groupId>org.netbeans.api</groupId>
-            <artifactId>org-netbeans-api-annotations-common</artifactId>
```
Comment (ilgrosso) [inline ide/netbeans/pom.xml]: `artifactId` removed??!?

**Comment by ilgrosso** [comment]
> @ilgrosso you weren't at the irc today, I have some questions about the thread safe comment. Please contact me when you're on.

When I am not in IRC, just add a comment here or send an e-mail to dev@syncope.apache.org, for better off-line handling.

File: ide/netbeans/pom.xml (hunk 6 above)
Comment (tmess567) [inline ide/netbeans/pom.xml]: Re-added it.

File: ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/connector/ResourceConnector.java (hunk 4 above)
Comment (tmess567) [inline ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/connector/ResourceConnector.java]: I don't really understand! It seems to be.

Comment (ilgrosso) [inline ide/netbeans/src/main/java/org/apache/syncope/netbeans/plugin/connector/ResourceConnector.java]: Static code can be accessed by multiple concurrent threads.
I would:
 * declare a monitor object as class static field:
  ```java
private static final Object MONITOR = new Object();
  ```
 * put the whole if block within a `synchronized` block, like as
  ```java
    synchronized(MONITOR) {
        if (MAIL_TTEMPLATE_MANAGER_SERVICE == null) {
             UserProperties userProperties = getUserProperties();
             MAIL_TTEMPLATE_MANAGER_SERVICE = new MailTemplateManagerService(
                     userProperties.getUrl(), userProperties.getUserName(),
                     userProperties.getPassword());
         }
    }
     return MAIL_TTEMPLATE_MANAGER_SERVICE;
  ```

**Pull request closed without merging**